scripts/core/with-live-query.tsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import {IBaseRestApiResponse, ILiveQueryProps, IResourceChange, IRestApiResponse} from 'superdesk-api';
+import {fetchChangedResources} from './helpers/CrudManager';
+import {httpRequestJsonLocal} from './helpers/network';
+import {throttleAndCombineArray} from './itemList/throttleAndCombine';
+import {addWebsocketEventListener} from './notification/notification';
+import {getQueryFieldsRecursive, toElasticQuery} from './query-formatting';
+
+type IState<T extends IBaseRestApiResponse> = {loading: true} | IStateReady<T>;
+
+interface IStateReady<T extends IBaseRestApiResponse> {
+    data: IRestApiResponse<T>;
+    loading: false;
+}
+
+export class WithLiveQuery<T extends IBaseRestApiResponse> extends React.PureComponent<ILiveQueryProps<T>, IState<T>> {
+    private eventListenersToRemoveBeforeUnmounting: Array<() => void>;
+    private handleContentChangesThrottled: (changes: Array<IResourceChange>) => void;
+
+    constructor(props: ILiveQueryProps<T>) {
+        super(props);
+
+        this.state = {loading: true};
+
+        this.eventListenersToRemoveBeforeUnmounting = [];
+
+        this.fetchItems = this.fetchItems.bind(this);
+        this.handleContentChanges = this.handleContentChanges.bind(this);
+
+        this.handleContentChangesThrottled = throttleAndCombineArray(
```

Comment:
I think throttle with 300 is not going to work, it needs to wait longer when the response takes longer, and refreshing every 300ms is too often anyhow, maybe we should have it configurable.

Reply:
I'll make it a second then.

https://github.com/superdesk/superdesk-client-core/pull/3847#discussion_r623072075 this would resolve the issue with server response taking longer than throttle wait. If new events arrive and it is still loading, it would get throttled again, which works like a queue that would only be released after the long running request has finished.